packages/fiori/src/Timeline.ts

```diff
@@ -111,8 +197,105 @@ class Timeline extends UI5Element {
 			: Timeline.i18nBundle.getText(TIMELINE_ARIA_LABEL);
 	}
 
+	get showBusyIndicatorOverlay() {
+		return !this.growsWithButton && this.loading;
+	}
+
+	get growsOnScroll(): boolean {
+		return this.growing === TimeLineGrowingMode.Scroll;
+	}
+
+	get growingButtonIcon() {
+		return this.layout === TimelineLayout.Horizontal ? process : drillDown;
+	}
+
+	get growsWithButton(): boolean {
+		return this.growing === TimeLineGrowingMode.Button;
+	}
+
+	onAfterRendering() {
+		if (this.growsOnScroll) {
+			this.observeTimeLineEnd();
+		}
+	}
+
+	onEnterDOM() {
+		this.growingIntersectionObserver = this.getIntersectionObserver();
+	}
+
+	onExitDOM() {
+		if (this.growingIntersectionObserver) {
+			this.growingIntersectionObserver.disconnect();
+			this.growingIntersectionObserver = null;
+		}
+		this.timeLineEndObserved = false;
+	}
+
+	observeTimeLineEnd() {
+		if (!this.timeLineEndObserved) {
+			this.getIntersectionObserver().observe(this.timelineEndMarker);
+			this.timeLineEndObserved = true;
+		}
+	}
+
+	getIntersectionObserver(): IntersectionObserver {
+		if (!this.growingIntersectionObserver) {
+			this.growingIntersectionObserver = new IntersectionObserver(this.onIntersection.bind(this), {
+				root: document,
+				threshold: 1.0,
+			});
+		}
+
+		return this.growingIntersectionObserver;
+	}
+
+	onIntersection(entries: Array<IntersectionObserverEntry>) {
+		if (this.initialIntersection) {
+			this.initialIntersection = false;
+			return;
+		}
+
+		if (entries.some(entry => entry.isIntersecting)) {
+			debounce(this.loadMore.bind(this), GROWING_WITH_SCROLL_DEBOUNCE_RATE);
+		}
+	}
+
+	loadMore() {
+		this.fireDecoratorEvent("load-more");
+	}
+
+	_onLoadMoreKeydown(e: KeyboardEvent) {
+		if (isSpace(e)) {
+			e.preventDefault();
+			this._loadMoreActive = true;
+		}
+
+		if (isEnter(e)) {
+			this._onLoadMoreClick();
+			this._loadMoreActive = true;
+		}
+	}
+
+	_onLoadMoreKeyup(e: KeyboardEvent) {
+		if (isSpace(e)) {
+			this._onLoadMoreClick();
+		}
+		this._loadMoreActive = false;
+	}
+
+	onInvalidation(change: ChangeInfo) {
+		if (change.type === "property" && change.name === "growing") {
+			this.timeLineEndObserved = false;
+			this.getIntersectionObserver().disconnect();
+		}
+	}
+
+	_onLoadMoreClick() {
+		this.fireDecoratorEvent("load-more");
+	}
+
 	_onfocusin(e: FocusEvent) {
-		let target = e.target as ITimelineItem | ToggleButton;
+		let target = e.target as ITimelineItem | ToggleButton | HTMLElement;
```

Comment:
HTMLElement is stronger than the two, you can remove them if you have it. But why is this needed? Can't you just add `Button` if this is for the load more button?